libsel4platsupport/src/io.c

```diff
@@ -141,11 +141,11 @@ static void *sel4platsupport_map_paddr_with_page_size(sel4platsupport_io_mapper_
     size += offset;
 
     io_mapping_t *mapping = new_node(BYTES_TO_SIZE_BITS_PAGES(size, page_size_bits));
-    assert(mapping->num_pages << page_size_bits >= size);
     if (!mapping) {
-        ZF_LOGE("Failed to allocate node for %zu pages", mapping->num_pages);
+        ZF_LOGE("Failed to allocate node for %zu pages", BYTES_TO_SIZE_BITS_PAGES(size, page_size_bits));
```

Comment:
We could do this, then there less redundancy and the asserts become easier to understand:

```
    size_t num_pages = BYTES_TO_SIZE_BITS_PAGES(size, page_size_bits);
    io_mapping_t *mapping = new_node(num_pages);
    if (!mapping) {
        ZF_LOGE("Failed to allocate node for %zu pages", num_pages);
        return NULL;
    }
    assert(mapping->num_pages == num_pages);
```

Reply:
Agree with using the horrible `BYTES_TO_SIZE_BITS_PAGES` only once.

The assert seems useless, but if you change it to check num_pages, it's really pointless. Either remove it altogether, or leave the check as it is.

Reply:
Right, it seem useless in the end. I didn't realize that `new_node()` is a local helper function.

Reply:
Assertion is really useless, let me remove it